EnvironmentSimulator/Libraries/esminiLib/esminiLib.cpp

```diff
@@ -1139,25 +1071,25 @@ extern "C"
                 object_type = scenarioengine::Object::Type::VEHICLE;
             }
 
-            Vehicle *vehicle = nullptr;
+            Vehicle                       *vehicle = nullptr;
             scenarioengine::OSCBoundingBox bb;
-            bb.center_.x_ = bounding_box.center_.x_;
-            bb.center_.y_ = bounding_box.center_.y_;
-            bb.center_.z_ = bounding_box.center_.z_;
+            bb.center_.x_          = bounding_box.center_.x_;
+            bb.center_.y_          = bounding_box.center_.y_;
+            bb.center_.z_          = bounding_box.center_.z_;
             bb.dimensions_.height_ = bounding_box.dimensions_.height_;
             bb.dimensions_.length_ = bounding_box.dimensions_.length_;
             bb.dimensions_.width_  = bounding_box.dimensions_.width_;
 
             if (object_type == scenarioengine::Object::Type::VEHICLE)
             {
-                vehicle             = new Vehicle();
-                object_id           = player->scenarioEngine->entities_.addObject(vehicle, true);
-                vehicle->name_      = name;
-                vehicle->scaleMode_ = EntityScaleMode::MODEL_TO_BB;
-                vehicle->model_id_  = model_id;
-                vehicle->model3d_   = SE_Env::Inst().GetModelFilenameById(model_id);
-                vehicle->category_  = object_category;
-                vehicle->role_      = object_role;
+                vehicle               = new Vehicle();
+                object_id             = player->scenarioEngine->entities_.addObject(vehicle, true);
+                vehicle->name_        = name;
+                vehicle->scaleMode_   = static_cast<EntityScaleMode>(scale_mode);
+                vehicle->model_id_    = model_id;
+                vehicle->model3d_     = SE_Env::Inst().GetModelFilenameById(model_id);
+                vehicle->category_    = object_category;
+                vehicle->role_        = object_role;
                 vehicle->boundingbox_ = bb;
```

Comment:
Not sure if writing to the bounding box of the vehicle when scale mode is set to 1 will lead to unexpected behavior.

Reply:
Good point. For scale_mode 1 (BB_TO_MODEL) it's easier to use the original function, since bounding box is calculated by esmini. But of course, it can be specified in the new function as well. In which case the specified bounding box will be ignored, so it can be set to anything, e.g. {{0.0, 0.0, 0.0}, {0.0, 0.0, 0.0}}.
I suggest we add info in documentation. See other comments.